TM1py/Services/CellService.py

```diff
@@ -991,6 +1011,156 @@ def write_through_unbound_process(self, cube_name: str, cellset_as_dict: Dict, i
+        :return: Success: bool, Messages: list, ChangeSet: None
+        """
+
+        application_service = ApplicationService(self._rest)
+        process_service = ProcessService(self._rest)
+        cube_service = self.get_cube_service()
+
+        # Generate hash based on tm1-session-id and local-thread guarantee unique name
+        # ToDo: is this reliably unique?
+        unique_string = f"{self._rest.session_id}{threading.get_ident()}"
+        unique_hash = hashlib.sha256(unique_string.encode('utf-8')).hexdigest()[:8]
+
+        # Transform the cellset into a dictionary and export to CSV
+        csv_content = StringIO()
+        csv_writer = csv.writer(
+            csv_content,
+            delimiter=",",
+            quoting=csv.QUOTE_ALL)
+        csv_writer.writerows(
+            list(elements) + [value.replace('\r', '').replace('\n', '') if isinstance(value, str) else value]
+            for elements, value
+            in cellset_as_dict.items())
+
+        # Define the path and file name
+        filename = f'{unique_hash}.csv'
+
+        # Upload CSV to TM1 server using ApplicationService:
+        # Create a folder TM1py
+        tm1py_folder = FolderApplication(path='', name='TM1py')
+        if not application_service.exists(path='', application_type='FOLDER', name=tm1py_folder.name):
+            application_service.create(tm1py_folder)
+
+        # Update or create the CSV file inside the TM1py folder
+        application = DocumentApplication(
+            path=tm1py_folder.name,
+            name=filename,
+            content=csv_content.getvalue().encode('utf-8'))
+
+        application_service.create(application=application, private=True, **kwargs)
+        try:
+            # Retrieve the file ID to be used as data source in the TI process. Default folder is .\Externals
+            blob_filename = application_service.get_document(
+                path=tm1py_folder.name,
+                name=filename,
+                private=True).file_id
+
+            # Create a TI process to load CSV  data into the cube
+            process_name = f"}}tm1py.{unique_hash}"
+            process = self._create_blob_write_process(cube_name, process_name, blob_filename,
+                                                      cube_service.get_dimension_names(cube_name), increment,
+                                                      skip_non_updateable, sandbox_name)
+
+            # Call the TI process with result
+            # process_service.create(process)
+            success, status, log_file = process_service.execute_process_with_return(process=process)
+            if not success:
+                if status in ['HasMinorErrors']:
+                    raise TM1pyWritePartialFailureException([status], [log_file], 1)
+                else:
+                    raise TM1pyWriteFailureException([status], [log_file])
+
+        # delete application in TM1
+        finally:
+            application_service.delete(
```

Comment:
Could this be added as a default argument? Could come in handy for debugging.

Reply:
You mean whether or not we delete the blob?

Reply:
yeah right